tests/unit_tests/c_wrappers/res/enkf/test_run_path_creation.py

```diff
@@ -267,3 +267,31 @@ def test_that_data_file_sets_num_cpu(eclipse_data, expected_cpus):
     res_config = ResConfig("config.ert")
     ert = EnKFMain(res_config)
     assert int(ert.get_context()["<NUM_CPU>"]) == expected_cpus
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_runpath_substitution_remain_valid():
+    """
+    This checks that runpath substitution remain intact.
+    """
+    config_text = dedent(
+        """
+        NUM_REALIZATIONS 2
+        JOBNAME my_case%d
+        RUNPATH real_%d/iter_%d
+        FORWARD_MODEL COPY_DIRECTORY(<FROM>=<CONFIG_PATH>/config.ert, <TO>=<RUNPATH>/output)
+        """
+    )
+    Path("config.ert").write_text(config_text)
+
+    res_config = ResConfig("config.ert")
+    ert = EnKFMain(res_config)
+
+    run_context = ert.create_ensemble_context("prior", [True, True], iteration=0)
+    ert.createRunPath(run_context)
+
+    for i, runcontext in enumerate(run_context):
+        assert (
+            "real_" + str(i) + "/iter_0"
+            in Path(runcontext.runpath + "/jobs.json").read_text()
+        )
```

Comment:
Could consider making this into a snapshot test?

```suggestion
snapshot.assert_match(Path(run_context[0].runpath + "/jobs.json").read_text(), "jobs.json_realization_0`)
snapshot.assert_match(Path(run_context[1].runpath + "/jobs.json").read_text(), "jobs.json_realization_1`)

Reply:
This is a good idea, but at the moment a bit challenging as the file is constantly changing due to ert_pid and others.